state modifed update (#7097)
fishtown-analytics/docs.getdbt.com: website/docs/reference/node-selection/methods.md

These selectors can help you shorten run times by excluding unchanged nodes. Currently, no subselectors are available at this time, but that might change as use cases evolve. 

> #### `state:modified`: node and reference impacts

Reviewer: ```suggestion
#### `state:modified` node and reference impacts
```

> `state:modified` also identifies any new nodes added, changes to existing nodes, and any changes made to:

Reviewer: ```suggestion
`state:modified` identifies any new nodes added, changes to existing nodes, and any changes made to:
```

- [access](/reference/resource-configs/access) permissions
- [`deprecation_date` ](/reference/resource-properties/deprecation_date)
- [`latest_version` ](/reference/resource-properties/latest_version)

If a node changes its group, downstream references may break, potentially causing build failures.

> As `group` is a config, and configs are generally included in `state:modified` detection, modifying the group name everywhere it’s referenced, will flag those nodes as "modified".

Reviewer: ```suggestion
As `group` is a config, and configs are generally included in `state:modified` detection, modifying the group name everywhere it’s referenced will flag those nodes as "modified".
```

> Depending on whether partial parsing is enabled, you will catch the breakage in CI.

Reviewer: ```suggestion
Depending on whether partial parsing is enabled, you will catch the breakage as part of CI workflows.
```

- If you change a group name everywhere it’s referenced, and partial parsing is enabled, dbt may only re-parse the changed model.
- If you update a group name in all its references without partial parsing enabled, dbt will re-parse all models and identify any invalid downstream references.

An error along the lines of “there’s nothing to do” can occur when you change the group name *and* something is picked up to be run via `dbt build --select state:modified`. This error will be caught at runtime so long as the CI job is selecting `state:modified+` (including downstreams).

> Certain factors can affect how references are used or resolved later on. These factors include:

Reviewer: ```suggestion
Certain factors can affect how references are used or resolved later on, including:
```

- Modifying access: if permissions or access rules change, some references might stop working.
- Modifying `deprecation_date`: if a reference or model version is marked  deprecated, new warnings might appear that affect how references are  processed.
- Modifying `latest_version`: if there’s no tie to a specific version, the reference or model will point to the latest version.

>   -  So if a newer version is released, the reference will automatically resolve to the new version, potentially changing the behavior or output of the system that relies on it.

Reviewer: ```suggestion
  -  If a newer version is released, the reference will automatically resolve to the new version, potentially changing the behavior or output of the system that relies on it.
```
